library/src/com/google/maps/android/geoJsonLayer/GeoJsonLayer.java

```diff
@@ -0,0 +1,253 @@
+
+    private GeoJsonLineStringStyle mDefaultLineStringStyle;
+
+    private GeoJsonPolygonStyle mDefaultPolygonStyle;
+
+    private boolean mGeoJsonDataOnMap;
+
+    private JSONObject mGeoJsonFile;
+
+    private GeoJsonRenderer mRenderer;
+
+    /**
+     * Creates a new Collection object
+     *
+     * @param map           GoogleMap object
+     * @param geoJsonObject JSONObject to parse GeoJSON data from
+     */
+    public GeoJsonLayer(GoogleMap map, JSONObject geoJsonObject) {
+        mFeatures = new HashMap<GeoJsonFeature, Object>();
+        mDefaultPointStyle = new GeoJsonPointStyle();
+        mDefaultLineStringStyle = new GeoJsonLineStringStyle();
+        mDefaultPolygonStyle = new GeoJsonPolygonStyle();
+        mGeoJsonDataOnMap = false;
+        mGeoJsonFile = geoJsonObject;
+        mRenderer = new GeoJsonRenderer(mFeatures, map);
+    }
+
+    /**
+     * Creates a new Collection object
+     *
+     * @param map        GoogleMap object
+     * @param resourceId Raw resource GeoJSON file
+     * @param context    Context object
+     * @throws IOException   if the file cannot be open for read
+     * @throws JSONException if the JSON file has invalid syntax and cannot be parsed successfully
+     */
+    public GeoJsonLayer(GoogleMap map, int resourceId, Context context)
+            throws IOException, JSONException {
+        mFeatures = new HashMap<GeoJsonFeature, Object>();
+        mDefaultPointStyle = new GeoJsonPointStyle();
+        mDefaultLineStringStyle = new GeoJsonLineStringStyle();
+        mDefaultPolygonStyle = new GeoJsonPolygonStyle();
+        mGeoJsonDataOnMap = false;
+        InputStream stream = context.getResources().openRawResource(resourceId);
+        mGeoJsonFile = createJsonFileObject(stream);
+        mRenderer = new GeoJsonRenderer(mFeatures, map);
+    }
+
+    /**
+     * Takes a character input stream and converts it into a JSONObject
+     *
+     * @param stream Character input stream representing  the GeoJSON file
+     * @return JSONObject representing the GeoJSON file
+     * @throws java.io.IOException    if the file cannot be opened for read
+     * @throws org.json.JSONException if the JSON file has poor structure
+     */
+    private JSONObject createJsonFileObject(InputStream stream) throws IOException, JSONException {
+        String line;
+        StringBuilder result = new StringBuilder();
+        // Reads from stream
+        BufferedReader reader = new BufferedReader(new InputStreamReader(stream));
+
+        // Read each line of the GeoJSON file into a string
+        while ((line = reader.readLine()) != null) {
+            result.append(line);
+        }
+
+        // Converts the result string into a JSONObject
+        return new JSONObject(result.toString());
+    }
+
+    /**
+     * Gets an iterator of all feature elements.
+     *
+     * @return iterator of feature elements
+     */
+    public java.util.Iterator<GeoJsonFeature> getFeatures() {
```

Comment:
returning Iterable might be more useful than returning Iterator because then developers can use it in for-each loops easily